draft-irtf-cfrg-opaque.md

```diff
@@ -1684,19 +1700,121 @@ protocols such as TLS.
+  of the derivation of OPRF keys via a single OPRF. As long as the derivation
+  of different OPRF keys from a single OPRF have different PRF inputs, the
+  protocol is secure. The choice of such inputs is up to the application.
+
+The following list enumerates notable differences and refinements from the original
+cryptographic design in {{OPAQUE}} and the corresponding CFRG document
+{{I-D.krawczyk-cfrg-opaque-03}} that were made to make this specification
+suitable for interoperable implementations.
+
+- {{OPAQUE}} used a generic prime-order group for the DH-OPRF and HMQV operations,
+  and includes necessary prime-order subgroup checks when receiving attacker-controlled
+  values over the wire. This specification instantiates the prime-order group using for
+  3DH using prime-order groups based on elliptic curves, as described in
+  {{I-D.irtf-cfrg-voprf, Section 2.1}}. This specification also delegates OPRF group
+  choice and operations to {{!I-D.irtf-cfrg-voprf}}. As such, the prime-order group as used
+  in the OPRF and 3DH as specified in this document both adhere to the requirements as
+  {{OPAQUE}}.
+- {{OPAQUE}} specified DH-OPRF (see Appendix B) to instantiate
+  the OPRF functionality in the protocol. A critical part of DH-OPRF is the
+  hash-to-group operation, which was left undefined in the original analysis.
+  However, the requirements for this operation were included. This specification
+  instantiates the OPRF functionality based on the {{I-D.irtf-cfrg-voprf}}, which
+  is identical to the DH-OPRF functionality in {{OPAQUE}} and, concretely, uses
+  the hash-to-curve functions in {{?I-D.irtf-cfrg-hash-to-curve}}. All hash-to-curve
+  methods in {{I-D.irtf-cfrg-hash-to-curve}} are compliant with the requirement
+  in {{OPAQUE}}, namely, that the output be a member of the prime-order group.
+- {{OPAQUE}} and {{I-D.krawczyk-cfrg-opaque-03}} both used HMQV as the AKE
+  for the protocol. However, this document fully specifies 3DH instead of HMQV
+  (though a sketch for how to instantiate OPAQUE using HMQV is included in {{hmqv-sketch}}).
+  Since 3DH satisfies the essential requirements for the AKE as described in {{OPAQUE}}
+  and {{I-D.krawczyk-cfrg-opaque-03}}, as described in {{security-analysis}}, this change
+  has no effect on the overall security of the protocol. 3DH was chosen for its
+  simplicity and ease of implementation.
+- The DH-OPRF and HMQV instantiation of OPAQUE in {{OPAQUE}}, Figure 12 uses
+  a different transcript than that which is described in this specification. In particular,
+  the key exchange transcript specified in {{ake-protocol}} is a superset of the transcript
+  as defined in {{OPAQUE}}. This was done to align with best practices, such as is
+  done for key exchange protocols like TLS 1.3 {{RFC8446}}.
+- Neither {{OPAQUE}} nor {{I-D.krawczyk-cfrg-opaque-03}} included wire format details for the
+  protocol, which is essential for interoperability. This specification rectifies this
```

Comment:
Replace 'rectifies' with 'fills'
(I was very explicit about these essential details not being in scope in my i-d, so I object to 'rectify' :-)

Reply:
Hah, fair :)